jsx/StaticDataTable.js

```diff
@@ -193,14 +267,8 @@ StaticDataTable = React.createClass({
                     data = this.props.Data[index[i].RowIdx][j];
                 }
 
-                // Increase counter, if filter value is found to be a substring
-                // of one of the column values
-                var filterData = this.props.Filter[this.props.Headers[j]];
-                if (filterData !== null &&
-                    data !== null &&
-                    data.indexOf(filterData) > -1
-                ) {
-                    filterMatchCount++;
+                if (this.hasFilterKeyword(this.props.Headers[j], data)) {
```

Comment:
@driusan It compares data at a specific column, to a filter of a specific column.

So, if filter is 
```js
var Filter = {
  'Name': 'Dave',
}
```
and your table is 

Name | State 
--- | --- 
Dave | CaDAVEr

It will only compare data that has same header in Filter and in DataTable, i.e the only column that will result in a match is `Name`. Hope it is more clear now.